llvm_zluda/src/lib.cpp

```diff
@@ -216,4 +208,47 @@ void LLVMZludaSetAtomic(
     }
 }
 
+std::mutex lld_mutex;
+
+int LLVMZludaLinkWithLLD(const char *input_path, const char *output_path, char **ErrorMessage)
+{
+    std::vector<const char *> args;
+    args.push_back("ld.lld");
+    args.push_back("-shared");
+    args.push_back(input_path);
+    args.push_back("-o");
+    args.push_back(output_path);
+    args.push_back("--threads=1");
+
+    std::string log_out_str;
+    std::string log_err_str;
+    llvm::raw_string_ostream log_out(log_out_str);
+    llvm::raw_string_ostream log_err(log_err_str);
+
+    // We don't want to leak memory as ZLUDA is running, so we destroy the linker context after every call, and wrap the invocation in a mutex.
+    std::lock_guard<std::mutex> guard(lld_mutex);
+    lld::Result result = lld::lldMain(args, log_out, log_err,
+                                      {{lld::Gnu, &lld::elf::link}});
+
+    lld::CommonLinkerContext::destroy();
```

Comment:
DO we really need this line?
It looks bit disconnected from the rest of the code and looking at LLD sources: https://github.com/llvm/llvm-project/blob/630dfc928c1e0668c3c0d59b303bb968ba44d7dd/lld/Common/DriverDispatcher.cpp, `lldMain` calls the function by itself anyway